Replace profile tcl tests

greghudson opened the pull request:
Along with PR #1189, these commits purge the last bit of Tcl code from the tree.

The test program is a pretty straightforward translation of prof_test1 into C, with a small amount of cleanup of how we handle copies of test.ini.  Most notably, the second test in prof_test1 winds up rewriting the test2.ini copy of test.ini, and that subtly affects all of the later tests.  The corresponding test_multi_handle() copies test2.ini to test3.ini first.

The test program calls out the shell fairly liberally to simplify file manipulations, as did the Tcl tests.  There is also a `sleep(2)` in there.  I dislike sleep calls in the test suite, but it doesn't have much overall impact.  I might dig into that test case a little harder since it's not well-described in the Tcl test script.

The name t_profile isn't great, since there's already a test_profile in that directory (a generic harness rather than a collection of tests).  I couldn't come up with a better name, though.

frozencemetery reviewed (changes requested):
prof_test1 should be deleted in the second commit

src/Makefile.in:639 has an exclusion that can be removed

(The profile deps file also returns a hit, but presumably this'll be taken care of in the next `make depend`.)

greghudson commented on .github/workflows/build.yml:
```diff
@@ -32,7 +32,7 @@ jobs:
               run: |
                 sudo apt-get update -qq
-                sudo apt-get install -y bison dejagnu gettext keyutils ldap-utils libcmocka-dev libldap2-dev libkeyutils-dev libresolv-wrapper libsasl2-dev libssl-dev python3-kdcproxy python3-pip slapd tcl-dev tcsh
+                sudo apt-get install -y bison dejagnu gettext keyutils ldap-utils libcmocka-dev libldap2-dev libkeyutils-dev libresolv-wrapper libsasl2-dev libssl-dev python3-kdcproxy python3-pip slapd tcsh
```
It's removed in PR #1189; I just haven't merged it yet.

greghudson commented:
I convinced myself that the sleep() probably doesn't have an impact on the current profile library design and removed it.

I wrote a file-creation helper since I think the code reads slightly better that way.  system() is still used for more complicated file operations.

greghudson commented:
I dug into test_multi_handle(), aka test2 in the TCL tests, and renamed and rewrote it.

The test was introduced in commit b3410d48ba12ba973944c202876240bdb5af71b3 alongside a bugfix to preserve the SHARED flag when updating the underlying file data of a profile object.  Erroneously clearing the SHARED flag meant that the profile wouldn't be removed from g_shared_trees when it was freed, creating a dangling reference.  (It also meant the profile wouldn't be copied during read-write setup, but the dangling reference is more straightforward to test.)

I experimented with reverting the fix and running the Tcl test2 by itself under valgrind, and it didn't show any memory faults.  either it wasn't triggering a file data update (perhaps due to commit 8a3056e4865063e4b2c2dce9d3316ae5526839da) or the dangling reference wasn't created until the outer handle (named c) was released, delaying the fault until a later test.  The corresponding C test also didn't trigger a fault until later, with or without the `sleep(2)`.

Commit 53d6d0d64e96b84b57b7446a787e683312e2a529 allows us to force a file data update without sleeping by changing the file timestamp and temporarily reopening the file.  With that in mind, we can exercise the bug using only init and release operations.  I rewrote the test in this manner, with comments explaining how it exercises the bug, and verified that the test causes a memory fault under valgrind with the bug present.
